holoviews/core/dimension.py

```diff
@@ -1126,12 +1140,54 @@ def opts(self, options=None, **kwargs):
 
                 kwargs = {k:{identifier:v} for k,v in kwargs.items()}
 
-        if options is None and kwargs=={}:
-            deep_clone = self.map(lambda x: x.clone(id=None))
-        else:
-            deep_clone = self.map(lambda x: x.clone(id=x.id))
-        StoreOptions.set_options(deep_clone, options, **kwargs)
-        return deep_clone
+        obj = self
+        if options is None and kwargs == {}:
+            if clone:
+                obj = self.map(lambda x: x.clone(id=None))
+            else:
+                self.map(lambda x: setattr(x, 'id', None))
+        elif clone:
+            obj = self.map(lambda x: x.clone(id=x.id))
+        StoreOptions.set_options(obj, options, backend=backend, **kwargs)
```

Comment:
I think ``obj=self`` is really confusing here as it is not *accessed* until the line:

```python
StoreOptions.set_options(obj, options, backend=backend, **kwargs)
```

Which can now be expressed in a way I find much more intuitive as:

```python
StoreOptions.set_options(obj if clone else self, options, backend=backend, **kwargs)
```

Reply:
I'll also note that although support for the ``clone`` argument is reasonable, I would have preferred to see it in a separate PR.

Reply:
>I think obj=self is really confusing here as it is not accessed until the line:

Not quite sure what you mean, it's overridden in the conditional and used twice. I'd have to do this instead:

     StoreOptions.set_options(obj if clone else self, options, backend=backend, **kwargs)
     return obj if clone else self